### PR TITLE
ex16v2_ch4.py

### PR DESCRIPTION
**Pull request by amymhaddad**
Write a program that asks the user for their age and compare it to the legal driving age of sixteen. If the user is sixteen or older, then the program should display “You are old enough to legally drive.” If the user is under sixteen, the program should display “You are not old enough to legally drive.”

### CODE REVIEW

File: ex16v2_ch4.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+Created on Thu May 10 15:34:22 2018
+
+@author: amyhaddad
+"""
+
+#step1: state constant
+LEGAL_DRIVING_AGE = 16
+
+#step2: get input from user
+age = int(input("Enter your age: "))
```
Comment (paulghaddad) [inline ex16v2_ch4.py]: Good use of `int` as opposed to `float`!

File: ex16v2_ch4.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+#step3: set up 3 conditionals: if user provides invalid input, if 16 or older, or if younger than 16
+if age < 0:
```
Comment (paulghaddad) [inline ex16v2_ch4.py]: Nice initial check for valid ages.

File: ex16v2_ch4.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    print("Please enter a valid age.")
+
```
Comment (paulghaddad) [inline ex16v2_ch4.py]: No blank line needed.

File: ex16v2_ch4.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+elif age >= 16:
```
Comment (paulghaddad) [inline ex16v2_ch4.py]: To take this to the next level, I'd personally keep the valid age check separate from the legal driving age check as they are separate concepts or concerns. So something like this:

```
# Check valid age
if age < 0:
  ## invalid age

#  Check whether the person can drive
if age >= 16:
  ##
else:
 ##
```

**Comment by amymhaddad** [comment]
@paulghaddad I made some small updates: deleted extra line and made valid age check separate from legal age check.

File: ex16v2_ch4.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+#step1: state constant
+LEGAL_DRIVING_AGE = 16
+
+#step2: get input from user
+age = int(input("Enter your age: "))
+
+#step3: set up 3 conditionals: if user provides invalid input, if 16 or older, or if younger than 16
+if age < 0:
+    print("Please enter a valid age.")
```
Comment (paulghaddad) [inline ex16v2_ch4.py]: The `if` on line 16 is a separate test from the `if-else` combination on lines 18-21, so I'd recommend adding a blank line after line 17 to separate the two.

Comment (paulghaddad) [reply ex16v2_ch4.py]: @amymhaddad Apply the comment above and it will be approved.

**Comment by amymhaddad** [comment]
@paulghaddad This is updated!

**Pull request merged**